### PR TITLE
Issue 41 redux toolkit

### PR DESCRIPTION
**Pull request by calster00**
develop

## GitHub Board

**Issue link**

[#41 Redux toolkit ](https://github.com/ita-social-projects/dokazovi-fe/issues/41 )

**Story link**

[#11 Story]( )

## Code reviewers
- [ ] @github_username

## Summary of issue

- [ ]  Summary of issue
- [ ]  Summary of issue

## Summary of change

- [x]  Update store & reducers to use Redux Toolkit
- [x]  Create thunks for async operations
- [x]  Update components appropriately

### CODE REVIEW

File: src/store/rootReducer.ts [hunk 1]
```diff
@@ -1,17 +1,10 @@
...
-  direction: IDirectionState;
-}
-
-export default combineReducers({
+const rootReducer = combineReducers({
   main: mainReducer,
-  direction: directionReducer,
 });
+
+export type RootState = ReturnType<typeof rootReducer>
```
Comment (SergeOlabin) [inline src/store/rootReducer.ts:8]: pls add `Type` add the end
```suggestion
export type RootStateType = ReturnType<typeof rootReducer>
```

you can search for eslint rule for that, also

**Review by SergeOlabin — APPROVED** [review]

File: src/store/rootReducer.ts (hunk 1 above)
Comment (calster00) [reply src/store/rootReducer.ts]: done

**Review by calster00 — COMMENTED** [review]
added eslint rule for types

**Review by SergeOlabin — CHANGES_REQUESTED** [review]
resolve conflicts

**Comment by calster00** [comment]
merged latest develop, resolved conflicts

**Comment by calster00** [comment]
CI is passing

**Pull request merged**